src/server/api/routers/searchLocation.ts

```diff
@@ -0,0 +1,51 @@
+import { z } from "zod";
+import { createTRPCRouter, publicProcedure } from "../trpc";
+import { Client } from "@googlemaps/google-maps-services-js";
+const client = new Client({});
+const GOOGLE_MAPS_API_KEY = process.env.GOOGLE_MAPS_API_KEY || "";
+if (!GOOGLE_MAPS_API_KEY) {
+    console.log("No gmaps API key, please set it")
+}
+export const searchRouter = createTRPCRouter({
+    getLocation: publicProcedure
+        .input(z.object({ locationQuery: z.string() }))
+        .query(async ({ input }) => {
+            try {
+                const response = await client.textSearch({
+                    params: {
+                        query: input.locationQuery,
+                        key: GOOGLE_MAPS_API_KEY
+                    },
+                    timeout: 1000,
+                });
+                const results = response.data.results;
+
+                if (results.length === 0) {
```

Comment:
This should not be an error -> we will just return an empty list and the FE will return "no results found"

error should be if !results